### PR TITLE
Rust 1.53 clippy updates

### PR DESCRIPTION
**Pull request by bgurney-rh**

### CODE REVIEW

File: src/cachedev.rs [hunk 1]
```diff
@@ -855,7 +855,7 @@ mod tests {
 
                 assert_eq!(status.metadata_mode, CacheDevMetadataMode::Good);
 
-                assert_eq!(status.needs_check, false);
+                assert!(status.needs_check, "{}", false);
```
Comment (mulkieran) [inline src/cachedev.rs:858]: There's a general pattern to follow here:

```assert_eq!(<expression>, false)``` -> ```assert!(!<expression>)```.

That should be quite adequate here. You _can_ use the two additional fields for a panic message  if you want to, but that's more work than necessary.

**Review by mulkieran — CHANGES_REQUESTED** [review]

File: src/cachedev.rs (hunk 1 above)
Comment (bgurney-rh) [reply src/cachedev.rs]: Resolved.

**Comment by bgurney-rh** [comment]
I'm going to create a new version of this PR; still with 2 commits, but cleaned up.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Looks ready! Please close and open a new PR for final review.

**Pull request closed without merging**